testsuite/tests/typing-local/local.ml

```diff
@@ -1903,24 +1903,6 @@ let primloc x = let local_ y = Int32.add x 1l in Int32.to_int y
 val primloc : int32 -> int = <fun>
 |}]
 
-(* (&&) and (||) tail call on the right *)
-let testbool1 x =
-  let local_ b = not x in
-  (b || false) && true
-
-let testbool2 x =
-  let local_ b = not x in
-  true && (false || b)
-[%%expect{|
-val testbool1 : bool -> bool = <fun>
-Line 7, characters 20-21:
-7 |   true && (false || b)
-                        ^
-Error: This local value escapes its region
-  Hint: Cannot return local value without an explicit "local_" annotation
-|}]
-
-
```

Comment:
these tests no longer fail because of mode crossing. Instead of deleting them, we can find some other tests that shows that the two operators tail call on the right.

Reply:
How about:
```
let testbool1 f =
  let local_ r = ref 42 in
  (f r || false) && true

let testbool2 f =
  let local_ r = ref 42 in
  true && (false || f r)
```